src/pveclib/vec_char_ppc.h

```diff
@@ -262,6 +269,70 @@ vec_clzb (vui8_t vra)
   return (r);
 }
 
+/** \brief Vector Count Trailing Zeros Byte.
+ *
+ *  Count the number of trailing '0' bits (0-16) within each byte
+ *  element of a 128-bit vector.
+ *
+ *  For POWER9 (PowerISA 3.0B) or later use the Vector Count Trailing
+ *  Zeros Byte instruction <B>vctzb</B>. Otherwise use a sequence of
+ *  pre ISA 3.0 VMX instructions.
+ *  SIMDized count Trailing zeros inspired by:
+ *
+ *  Warren, Henry S. Jr and <I>Hacker's Delight</I>, 2nd Edition,
+ *  Addison Wesley, 2013. Chapter 5 Counting Bits, Section 5-4.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |  6-8  | 2/cycle  |
+ *  |power9   |   3   | 2/cycle  |
+ *
+ *  @param vra 128-bit vector treated as 16 x 8-bit char
+ *  (byte) elements.
+ *  @return 128-bit vector with the Trailng Zeros count for each
+ *  byte element.
+ */
+static inline vui8_t
+vec_ctzb (vui8_t vra)
+{
+  vui8_t r;
+#ifdef _ARCH_PWR9
+#if defined (vec_cnttz) || defined (__clang__)
```

Comment:
What's the `__clang__` about? Does clang universally support this on POWER9?

Reply:
If _ARCH_PWR9 is defined then Clang also supports a generic vec_cnttz built-in (but not a macro like GCC). 
If I had to explain all bugs and weirdness for every case I would never finish :-/

Reply:
Would it be better to use `__has_builtin`?

Reply:
There is no documented __has_builtin for GCC.

Reply:
https://gcc.gnu.org/onlinedocs/gcc-10.1.0/cpp/_005f_005fhas_005fbuiltin.html